README.md

```diff
@@ -593,6 +604,35 @@ for promises that are not already resolved by the time you attempt to flush them
 
 </details>
 
+<details>
+
+<summary><a name="waitForExcept"></a>Is there a different way to wait for things to happen? For example for end to end or contract tests?</summary>
+Definitely! There is an abstraction called `waitForExpect` that will keep
+calling your expectations until a timeout or the expectation passes - whatever happens first.
+
+Please take a look at this example (taken from [`here`](https://github.com/kentcdodds/react-testing-library/blob/master/src/__tests__/end-to-end.js)):
+
+```javascript
+import {render, waitForExpect} from 'react-testing-library'
+test('it waits for the data to be loaded', async () => {
+  const {queryByText, queryByTestId} = render(<ComponentWithLoader />)
+
+  // Initially the loader shows
+  expect(queryByText('Loading...')).toBeTruthy()
+
+  // This will pass when the state of the component changes once the data is available
+  // the loader will disappear, and the data will be shown
+  await waitForExpect(() => {
+    expect(queryByText('Loading...')).toBeNull()
+    expect(queryByTestId('message').textContent).toMatch(/Hello World/)
+  })
```

Comment:
Similar to my comment in the test, let's update this so only the necessary assertions are inside the callback 👍